agent/repowise.py

```
@@ -0,0 +1,233 @@
+                    relevant_hotspots.append({"path": f, "changes": hotspots[f], "risk": "high"})
+
+        return {
+            "overall_hotspots": self.get_hotspots(limit=10),
+            "impact_analysis": relevant_hotspots
+        }
+
+    def get_why(self, target: str) -> str:
+        """Extracts architectural decisions related to target from git history."""
+        try:
+            cmd = f"git log --grep='WHY' --grep='DECISION' --grep='TRADEOFF' -i -- '{target}'"
+            result = subprocess.run(cmd, shell=True, cwd=self.root, capture_output=True, text=True)
```

Ruff (0.15.12)

[error] `subprocess` call with `shell=True` identified, security issue

(S602)